### PR TITLE
Revise CI for pyarrow/conda support

### PR DESCRIPTION
**Pull request by akarve**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,72 @@
...
+  - linux
+  - osx
+  # TODO: windows CI (not currently supported by travis)
+
+env:
+  global:
+    - USE_CONDA=true
+  matrix:
+    - QPYVER=2.7
+    - QPYVER=3.4
```
Comment (dimaryaz) [inline .travis.yml]: Why not just remove it from here? You want to keep supporting 3.4 on Linux?

File: .travis.yml [hunk 2]
```diff
@@ -1,42 +1,72 @@
...
+  allow_failures:
+    # no wheel support, so pyarrow installation fails
+    - env: TRAVIS_OS_NAME=osx, QPYVER=3.4
+  fast_finish: true
   include:
     - os: linux
+      language: python
+      python: "2.7"
+      env: USE_CONDA=false
       sudo: required
```
Comment (dimaryaz) [inline .travis.yml]: I suspect that `sudo` is not required anymore? Though I guess we might need it later.

File: .travis.yml [hunk 3]
```diff
@@ -1,42 +1,72 @@
...
-  # travis_retry here because occasionally fails on high latency
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then travis_retry brew update; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install homebrew/science/hdf5@1.8; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export HDF5_DIR=$(brew --prefix homebrew/science/hdf5@1.8); fi
-  # apt-get libhdf5-dev doesn't work in spite of https://github.com/travis-ci/apt-package-whitelist/issues/416
-  # see also: https://github.com/travis-ci/apt-package-whitelist/issues/416
-  # right now this looks like the best HDF5 bet
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install libhdf5-serial-dev; fi
-  - echo $(python --version)
+  - if [[ "$USE_CONDA" = true ]]; then
```
Comment (dimaryaz) [inline .travis.yml]: `==` and `"true"`, just for consistency with the other `if`s.

File: .travis.yml [hunk 4]
```diff
@@ -1,42 +1,72 @@
...
+        curl -o mconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh;
+      fi
+      chmod +x mconda.sh;
+      ./mconda.sh -b;
+      export PATH="$HOME/miniconda3/bin:$PATH";
+    fi
 
 install:
+  - if [[ "$USE_CONDA" = true ]]; then
+      conda create --yes -n quilt_env python=$QPYVER pip;
```
Comment (dimaryaz) [inline .travis.yml]: `"$QPYVER"`

File: .travis.yml [hunk 5]
```diff
@@ -1,42 +1,72 @@
...
+      if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+        curl -o mconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh;
+      fi
+      chmod +x mconda.sh;
+      ./mconda.sh -b;
+      export PATH="$HOME/miniconda3/bin:$PATH";
+    fi
 
 install:
+  - if [[ "$USE_CONDA" = true ]]; then
```
Comment (dimaryaz) [inline .travis.yml]: Same as above - `==` and `"true"`

File: .travis.yml [hunk 6]
```diff
@@ -1,42 +1,72 @@
...
+      conda create --yes -n quilt_env python=$QPYVER pip;
+      conda install --yes -c conda-forge pyarrow=0.3;
+      source activate quilt_env;
+    fi
+  # debug support
+  - echo $(python --version)
```
Comment (dimaryaz) [inline .travis.yml]: Just do `python --version` - no need for `echo $(...)`

File: .travis.yml (hunk 1 above)
Comment (akarve) [inline .travis.yml]: yeah i made 3.4 a non-block test on `osx` as well

File: .travis.yml (hunk 2 above)
Comment (akarve) [inline .travis.yml]: can try removing; was getting permissions errors

File: .travis.yml (hunk 6 above)
Comment (akarve) [inline .travis.yml]: CI treats them differently as I recall; can try

**Comment by akarve** [comment]
TODO
* Update README installation and supported platforms

File: .travis.yml (hunk 2 above)
Comment (akarve) [inline .travis.yml]: i checked and `sudo` is in fact required

File: .travis.yml [hunk 7]
```diff
@@ -1,42 +1,74 @@
...
+      fi
+      chmod +x mconda.sh
+      ./mconda.sh -b
+      export PATH="$HOME/miniconda3/bin:$PATH"
+    fi
 
 install:
+  - |
+    if [[ "$USE_CONDA" = true ]]; then
+      conda create --yes -n quilt_env python=$QPYVER pip;
```
Comment (dimaryaz) [inline .travis.yml]: Don't need `;`s

File: .travis.yml [hunk 8]
```diff
@@ -1,42 +1,74 @@
...
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then travis_retry brew update; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install homebrew/science/hdf5@1.8; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export HDF5_DIR=$(brew --prefix homebrew/science/hdf5@1.8); fi
-  # apt-get libhdf5-dev doesn't work in spite of https://github.com/travis-ci/apt-package-whitelist/issues/416
-  # see also: https://github.com/travis-ci/apt-package-whitelist/issues/416
-  # right now this looks like the best HDF5 bet
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install libhdf5-serial-dev; fi
-  - echo $(python --version)
+  - |
+    if [[ "$USE_CONDA" = true ]]; then
```
Comment (dimaryaz) [inline .travis.yml]: `... == "true"`

File: .travis.yml [hunk 9]
```diff
@@ -1,42 +1,74 @@
...
+        curl -o mconda.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+      fi
+      chmod +x mconda.sh
+      ./mconda.sh -b
+      export PATH="$HOME/miniconda3/bin:$PATH"
+    fi
 
 install:
+  - |
+    if [[ "$USE_CONDA" = true ]]; then
```
Comment (dimaryaz) [inline .travis.yml]: `... == "true"`

**Comment by dimaryaz** [comment]
@akarve: I'll merge this so that CI works for my other PR. We can tweak things later.

**Pull request merged**